webapp/src/dialogs.tsx

```diff
@@ -596,17 +597,32 @@ export function showCreateGithubRepoDialogAsync(name?: string) {
         }
     }
 
+    function onPublicChanged(v: boolean) {
+        v = !!v;
+        if (repoPublic != v) {
+            repoPublic = v;
+            coretsx.forceUpdate();
+        }
+    }
+
     return core.confirmAsync({
+        hideCancel: true,
+        hasCloseIcon: true,
         header: lf("Create GitHub repository"),
         jsxd: () => {
             const nameErr = repoNameError();
             return <div className="ui form">
+                <p>
+                    {lf("Host your code on GitHub and work together with friends on projects.")}
+                    <sui.Link href="https://github.com/about" target="_blank" icon="question circle" />
+                </p>
                 <div className="ui field">
-                    <sui.Input type="url" value={repoName} onChange={onNameChanged} label={lf("Repository name.")} placeholder={`pxt-my-gadget...`} class="fluid" error={nameErr} />
+                    <sui.Input type="url" value={repoName} onChange={onNameChanged} label={lf("Repository name")} placeholder={`pxt-my-gadget...`} class="fluid" error={nameErr} />
                 </div>
                 <div className="ui field">
-                    <sui.Input type="text" value={repoDescription} onChange={onDescriptionChanged} label={lf("Repository description.")} placeholder={lf("MakeCode extension for my gadget")} class="fluid" />
+                    <sui.Input type="text" value={repoDescription} onChange={onDescriptionChanged} label={lf("Repository description")} placeholder={lf("MakeCode extension for my gadget")} class="fluid" />
                 </div>
+                <sui.Checkbox checked={repoPublic} inputLabel={repoPublic ? lf("Public repository, anyone can look at your code.") : lf("Private repository, your code is only visible to you.")} onChange={onPublicChanged} />
```

Comment:
I think this may be confusing that the description changes when you check/uncheck, I've never seen that in a UI.